### ISSUE TITLE
Bean is not injected with "uses" attribute of @Mapper annotation

### ISSUE THREAD

**Issue by belkin1667** [mapstruct/mapstruct#2686]
I have the following `@Mapper` annotation with `LocationDetailsMapper` injection
```
@Mapper(
        componentModel = "spring",
        unmappedTargetPolicy = ReportingPolicy.ERROR,
        uses = {
                LocationDetailsMapper.class,
        }
)
```

I expect that I can use the following `@Mapping`:
```
@Mapping(target = "details", expression = "java(locationDetailsMapper.mapDetails(someSource1, someSource2))")
```

However, generated code does not have locationDetailsMapper injection

The only way to have mapper injected to use it as in the snippet above is to provide another `@Mapping(target = "smth", source = "someSource1.smth")`, which would use that LocationDetailsMapper method like `OtherMyClass map(MyClass smth)` somehow and only then I can use expressions like `expression = "java(locationDetailsMapper.mapDetails(someSource1, someSource2))"`, which is a bit unexpected behavior - as I thought if I specify something in `uses` I expect it to be injected unconditionally

To sum up - mapper is injected only in case mapstruct can automatically detect where to use that mapper and if it can not do so it does not inject the specified mapper (which is exactly my case when I want to specify my custom mapping with 2 or more arguments in `expression`)

**Comment by Zegveld** [comment]
Seems to be by design, see https://github.com/mapstruct/mapstruct/issues/2237#issuecomment-710776073

I'll mark this as a documentation update, because there it indeed says that it will be injected. That needs updated so it correctly defines this behavior.

Best way to handle it in your case it to make the mapper an abstract class with an `@AutoWired` protected field named `locationDetailsMapper`, then the injection behavior should work.
